packages/pkp-walletconnect/src/lib/pkp-walletconnect.ts

```diff
@@ -0,0 +1,638 @@
+    for (const pkpClient of this.pkpClients) {
+      let wallet: PKPBase;
+      let address: string;
+      switch (chainName) {
+        case 'eip155':
+          wallet = pkpClient.getEthWallet();
+          address = await wallet.getAddress();
+          addresses.push(formatAccountWithChain(address, chain));
+          break;
+        default:
+          break;
+      }
+    }
+    return addresses;
+  }
+
+  /**
+   * Check if chain is supported by Lit
+   *
+   * @param {string} chain - Chain in CAIP-2 format
+   *
+   * @returns {boolean} - True if chain is supported, false otherwise
+   */
+  public checkIfChainIsSupported(chain: string): boolean {
+    const parsedChain = parseChainId(chain);
+    const chainId = Number.parseInt(parsedChain.reference, 10);
+    // TODO: Update this once we support more JSON RPC handlers
+    if (parsedChain.namespace === 'eip155') {
+      for (const key in LIT_CHAINS) {
+        const chain = LIT_CHAINS[key];
+        if (chain.chainId === chainId) {
+          return true;
+        }
+      }
+      return false;
+    } else {
+      return false;
+    }
+  }
+
+  /**
+   * Find PKPClient by request event params
+   *
+   * @param {any} params - Request event params
+   */
+  public async findPKPClientByRequestParams(
+    params: any
+  ): Promise<PKPClient | null> {
+    const paramsString = JSON.stringify(params);
+
+    // Loop through all wallets and find the one that has an address that can be found within the request params
+    for (const pkpClient of this.pkpClients) {
+      // TODO: Update this once we support more JSON RPC handlers
+      const ethWallet = pkpClient.getEthWallet();
+      const ethAddress = await ethWallet.getAddress();
+      if (paramsString.toLowerCase().includes(ethAddress.toLowerCase())) {
+        return pkpClient;
+      }
+    }
+    return null;
+  }
+
+  /**
+   * Add a PKPClient to list of PKPClients if not already added
+   *
+   * @param {PKPClient} pkpClient - The PKPClient instance
+   */
+  public addPKPClient(pkpClient: PKPClient): void {
+    const existingClient = this.pkpClients.find(
+      (client) => client.pkpPubKey === pkpClient.pkpPubKey
+    );
+    if (!existingClient) {
+      this.pkpClients.push(pkpClient);
+    }
+  }
+
+  /**
+   * Get current list of PKPClients
+   *
+   * @returns {PKPClient[]} - List of PKPClients
+   */
+  public getPKPClients(): PKPClient[] {
+    return this.pkpClients;
+  }
+
+  // ----------------- Private methods -----------------
+
+  /**
+   * Checks if the given WalletConnect client is initialized and returns it. If it's not initialized, throws an error.
+   *
+   * @private
+   * @param {IWeb3Wallet | undefined} client - The WalletConnect client instance to check for initialization
+   * @returns {IWeb3Wallet} - The initialized WalletConnect client instance
+   * @throws {Error} - If the WalletConnect client instance is not initialized
+   */
+  private _isWalletConnectInitialized(
```

Comment:
I thought the only way to do private class methods was to prefix with `#`? I.e
```suggestion
  #isWalletConnectInitialized(
```

Reply:
also usually methods with `is` return a boolean - since we're returning a client object here, maybe we should rename the method?

Reply:
in typescript, one can use the `private` keyword to declare a private method